components/style/values/computed/mod.rs

```diff
@@ -169,3 +170,73 @@ impl ToCss for BorderRadiusSize {
 
 pub type Number = CSSFloat;
 pub type Opacity = CSSFloat;
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub enum AngleOrCorner {
+    Angle(Angle),
+    Corner(HorizontalDirection, VerticalDirection),
+}
+
+impl ToComputedValue for specified::AngleOrCorner {
+    type ComputedValue = AngleOrCorner;
+
+    #[inline]
+    fn to_computed_value(&self, _: &Context) -> AngleOrCorner {
+        match *self {
+            specified::AngleOrCorner::Angle(angle) => {
+                AngleOrCorner::Angle(angle)
+            },
+            specified::AngleOrCorner::Corner(horizontal, vertical) => {
+                match (horizontal, vertical) {
+                    (None, Some(VerticalDirection::Top)) =>
+                        AngleOrCorner::Angle(Angle(0.0)),
+                    (Some(HorizontalDirection::Right), Some(VerticalDirection::Top)) =>
+                        AngleOrCorner::Angle(Angle(PI * 0.25)),
+                    (Some(HorizontalDirection::Right), None) =>
+                        AngleOrCorner::Angle(Angle(PI * 0.5)),
+                    (Some(HorizontalDirection::Right), Some(VerticalDirection::Bottom)) =>
+                        AngleOrCorner::Angle(Angle(PI * 0.75)),
+                    (None, Some(VerticalDirection::Bottom)) =>
+                        AngleOrCorner::Angle(Angle(PI)),
+                    (Some(HorizontalDirection::Left), Some(VerticalDirection::Bottom)) =>
```

Comment:
Thanks for helping me work through this stuff.  I went ahead and updated the Match block.

Looks like the test is still unhappy here.  From what I can tell, I think that `specified::AngleOrCorner.to_computed_value()` isn't being called. I'm not really sure where to look to debug that further.

```
---- parsing::image::test_linear_gradient stdout ----
        thread 'parsing::image::test_linear_gradient' panicked at 'assertion failed: `(left == right)` (left: `"linear-gradient(to left, red, green)"`, right: `"linear-gradient(4.712389rad, red, green)"`)', /Users/nprice/Projects/servo/tests/unit/style/parsing/image.rs:15
```